server/src/main/java/org/opensearch/node/Node.java

```diff
@@ -812,7 +813,7 @@ protected Node(
                 remoteClusterStateCleanupManager = null;
             }
             final RemoteStorePinnedTimestampService remoteStorePinnedTimestampService;
-            if (isRemoteStoreAttributePresent(settings)) {
+            if (isRemoteStoreAttributePresent(settings) && CLUSTER_REMOTE_STORE_PINNED_TIMESTAMP_ENABLED.get(settings)) {
```

Comment:
Why not use `RemoteStoreSettings.isPinnedTimestampsEnabled()` here?

Reply:
Because `RemoteStoreSettings.isPinnedTimestampsEnabled()` requires `RemoteStoreSettings` to be initialised. Here, `RemoteStoreSettings` is initialised after this code block.